<?xml version="1.0" ?>
<robot name="dh_ag95_gripper_standalone">
    
    <!-- Define planning groups -->
    <group name="gripper">
        <joint name="gripper_finger1_joint" />
        <joint name="gripper_finger2_joint" />
        <joint name="gripper_finger1_finger_joint" />
        <joint name="gripper_finger2_finger_joint" />
        <joint name="gripper_finger1_inner_knuckle_joint" />
        <joint name="gripper_finger2_inner_knuckle_joint" />
        <joint name="gripper_finger1_finger_tip_joint" />
        <joint name="gripper_finger2_finger_tip_joint" />
    </group>
    
    <!-- Define passive joints -->
    <passive_joint name="gripper_finger1_inner_knuckle_joint" />
    <passive_joint name="gripper_finger1_finger_tip_joint" />
    <passive_joint name="gripper_finger2_joint" />
    <passive_joint name="gripper_finger2_inner_knuckle_joint" />
    <passive_joint name="gripper_finger2_finger_tip_joint" />
    <passive_joint name="gripper_finger1_finger_joint" />
    <passive_joint name="gripper_finger2_finger_joint" />
    
    <!-- Define group states -->
    <group_state name="open" group="gripper">
        <joint name="gripper_finger1_joint" value="0.0" />
    </group_state>
    
    <group_state name="closed" group="gripper">
        <joint name="gripper_finger1_joint" value="0.5" />
    </group_state>
    
    <!-- Disable collisions between base and all finger links -->
    <disable_collisions link1="gripper_base_link" link2="gripper_finger1_finger_link" reason="Never" />
    <disable_collisions link1="gripper_base_link" link2="gripper_finger1_finger_tip_link" reason="Never" />
    <disable_collisions link1="gripper_base_link" link2="gripper_finger1_inner_knuckle_link" reason="Adjacent" />
    <disable_collisions link1="gripper_base_link" link2="gripper_finger1_knuckle_link" reason="Adjacent" />
    <disable_collisions link1="gripper_base_link" link2="gripper_finger2_finger_link" reason="Never" />
    <disable_collisions link1="gripper_base_link" link2="gripper_finger2_finger_tip_link" reason="Never" />
    <disable_collisions link1="gripper_base_link" link2="gripper_finger2_inner_knuckle_link" reason="Adjacent" />
    <disable_collisions link1="gripper_base_link" link2="gripper_finger2_knuckle_link" reason="Adjacent" />
    
    <!-- Disable collisions between finger 1 components -->
    <disable_collisions link1="gripper_finger1_finger_link" link2="gripper_finger1_finger_tip_link" reason="Adjacent" />
    <disable_collisions link1="gripper_finger1_finger_link" link2="gripper_finger1_inner_knuckle_link" reason="Never" />
    <disable_collisions link1="gripper_finger1_finger_link" link2="gripper_finger1_knuckle_link" reason="Adjacent" />
    <disable_collisions link1="gripper_finger1_finger_tip_link" link2="gripper_finger1_inner_knuckle_link" reason="Never" />
    <disable_collisions link1="gripper_finger1_finger_tip_link" link2="gripper_finger1_knuckle_link" reason="Never" />
    <disable_collisions link1="gripper_finger1_inner_knuckle_link" link2="gripper_finger1_knuckle_link" reason="Never" />
    
    <!-- Disable collisions between finger 2 components -->
    <disable_collisions link1="gripper_finger2_finger_link" link2="gripper_finger2_finger_tip_link" reason="Adjacent" />
    <disable_collisions link1="gripper_finger2_finger_link" link2="gripper_finger2_inner_knuckle_link" reason="Never" />
    <disable_collisions link1="gripper_finger2_finger_link" link2="gripper_finger2_knuckle_link" reason="Adjacent" />
    <disable_collisions link1="gripper_finger2_finger_tip_link" link2="gripper_finger2_inner_knuckle_link" reason="Never" />
    <disable_collisions link1="gripper_finger2_finger_tip_link" link2="gripper_finger2_knuckle_link" reason="Never" />
    <disable_collisions link1="gripper_finger2_inner_knuckle_link" link2="gripper_finger2_knuckle_link" reason="Never" />
    
    <!-- Disable collisions between finger 1 and finger 2 components -->
    <disable_collisions link1="gripper_finger1_finger_link" link2="gripper_finger2_finger_link" reason="Never" />
    <disable_collisions link1="gripper_finger1_finger_link" link2="gripper_finger2_finger_tip_link" reason="Never" />
    <disable_collisions link1="gripper_finger1_finger_link" link2="gripper_finger2_inner_knuckle_link" reason="Never" />
    <disable_collisions link1="gripper_finger1_finger_link" link2="gripper_finger2_knuckle_link" reason="Never" />
    <disable_collisions link1="gripper_finger1_finger_tip_link" link2="gripper_finger2_finger_link" reason="Never" />
    <disable_collisions link1="gripper_finger1_finger_tip_link" link2="gripper_finger2_finger_tip_link" reason="Never" />
    <disable_collisions link1="gripper_finger1_finger_tip_link" link2="gripper_finger2_inner_knuckle_link" reason="Never" />
    <disable_collisions link1="gripper_finger1_finger_tip_link" link2="gripper_finger2_knuckle_link" reason="Never" />
    <disable_collisions link1="gripper_finger1_inner_knuckle_link" link2="gripper_finger2_finger_link" reason="Never" />
    <disable_collisions link1="gripper_finger1_inner_knuckle_link" link2="gripper_finger2_finger_tip_link" reason="Never" />
    <disable_collisions link1="gripper_finger1_inner_knuckle_link" link2="gripper_finger2_inner_knuckle_link" reason="Never" />
    <disable_collisions link1="gripper_finger1_inner_knuckle_link" link2="gripper_finger2_knuckle_link" reason="Never" />
    <disable_collisions link1="gripper_finger1_knuckle_link" link2="gripper_finger2_finger_link" reason="Never" />
    <disable_collisions link1="gripper_finger1_knuckle_link" link2="gripper_finger2_finger_tip_link" reason="Never" />
    <disable_collisions link1="gripper_finger1_knuckle_link" link2="gripper_finger2_inner_knuckle_link" reason="Never" />
    <disable_collisions link1="gripper_finger1_knuckle_link" link2="gripper_finger2_knuckle_link" reason="Never" />
    
    <!-- End effector definition -->
    <end_effector name="gripper_end_effector" parent_link="gripper_finger1_finger_tip_link" group="gripper" />
    <end_effector name="gripper_end_effector_2" parent_link="gripper_finger2_finger_tip_link" group="gripper" />
    
    <!-- Virtual joint for standalone operation -->
    <virtual_joint name="virtual_joint" type="fixed" parent_frame="world" child_link="gripper_base_link" />
    
</robot> 